src/Explorer/Panes/PublishNotebookPaneComponent.tsx

```diff
@@ -0,0 +1,209 @@
+import { ITextFieldProps, Stack, Text, TextField, Dropdown, IDropdownProps } from "office-ui-fabric-react";
+import * as React from "react";
+import { GalleryCardComponent } from "../Controls/NotebookGallery/Cards/GalleryCardComponent";
+import { FileSystemUtil } from "../Notebook/FileSystemUtil";
+import "./PublishNotebookPaneComponent.less";
+
+export interface PublishNotebookPaneProps {
+  notebookName: string;
+  notebookDescription: string;
+  notebookTags: string;
+  notebookAuthor: string;
+  onChangeDescription: (newValue: string) => void;
+  onChangeTags: (newValue: string) => void;
+  onChangeImageSrc: (newValue: string) => void;
+  onError: (formError: string, formErrorDetail: string, area: string) => void;
+  clearFormError: () => void;
+}
+
+interface PublishNotebookPaneState {
+  type: string;
+  notebookDescription: string;
+  notebookTags: string;
+  imageSrc: string;
+}
+
+export class PublishNotebookPaneComponent extends React.Component<PublishNotebookPaneProps, PublishNotebookPaneState> {
```

Comment:
Since this is a React component. Can we add a unit test for this? Search for `.toMatchSnapshot()`. At a minimum, you can shallow-mount the component with some bogus data and it'll test if the component is rendered properly.
If you're feeling adventurous, you can also simulate clicks and test with asserts as usual.

Reply:
will add in the next iteration